src/main/java/org/sonatype/repository/vgo/internal/VgoContentValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.vgo.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.mime.MimeRulesSource;
+import org.sonatype.nexus.repository.storage.ContentValidator;
+import org.sonatype.nexus.repository.storage.DefaultContentValidator;
+import org.sonatype.repository.vgo.VgoFormat;
+
+import com.google.common.base.Supplier;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Vgo specific {@link ContentValidator} that "hints" default content validator for some vgo specific file
+ * extensions and format specific files.
+ *
+ * @since 0.0.1
+ */
+@Named(VgoFormat.NAME)
+@Singleton
+public class VgoContentValidator
+    extends ComponentSupport
+    implements ContentValidator
+{
+  private final DefaultContentValidator defaultContentValidator;
+
+  @Inject
+  public VgoContentValidator(final DefaultContentValidator defaultContentValidator) {
+    this.defaultContentValidator = checkNotNull(defaultContentValidator);
+  }
+
+  @Nonnull
+  @Override
+  public String determineContentType(boolean strictContentTypeValidation,
+                                     Supplier<InputStream> contentSupplier,
+                                     @Nullable MimeRulesSource mimeRulesSource,
+                                     @Nullable String contentName,
+                                     @Nullable String declaredContentType) throws IOException
+  {
+    if (contentName != null) {
+      if (contentName.endsWith(".mod")) {
```

Comment:
probably if simplify into one line it could be more readable? ex.:
```suggestion
     if (contentName != null && contentName.endsWith(".mod")) {
```